tests/utils.go

```diff
@@ -2688,16 +2688,15 @@ func waitForVMIPhase(phases []v1.VirtualMachineInstancePhase, obj runtime.Object
 	virtClient, err := kubecli.GetKubevirtClient()
 	ExpectWithOffset(1, err).ToNot(HaveOccurred())
 
-	// In case we don't want errors, start an event watcher and  check in parallel if we receive some warnings
-	if ignoreWarnings != true {
+	vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &metav1.GetOptions{})
+	ExpectWithOffset(1, err).ToNot(HaveOccurred())
 
-		// Fetch the VirtualMachineInstance, to make sure we have a resourceVersion as a starting point for the watch
-		// FIXME: This may start watching too late and we may miss some warnings
-		if vmi.ResourceVersion == "" {
-			vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &metav1.GetOptions{})
-			ExpectWithOffset(1, err).ToNot(HaveOccurred())
-		}
+	if isPhaseIncludedIn(vmi.Status.Phase, phases) {
+		return
```

Comment:
You should initialize the nodeName before you return.

Reply:
Missed that one. thanks!